zrml/parimutuel/src/benchmarking.rs

```diff
@@ -134,6 +142,55 @@ mod benchmarks {
         claim_refunds(RawOrigin::Signed(loser_0), loser_0_asset);
     }
 
+    #[benchmark]
+    fn on_activation() {
+        let market_id = setup_market::<T>(MarketType::Categorical(64u16));
+
+        #[block]
+        {
+            Parimutuel::<T>::on_activation(&market_id);
+        }
+
+        for asset_idx in 0..64 {
+            let asset = ParimutuelShareOf::<T>::Share(Zero::zero(), asset_idx).into();
+            assert!(T::AssetCreator::asset_exists(asset));
+        }
+    }
+
+    #[benchmark]
+    fn on_resolution() {
+        let market_id = setup_market::<T>(MarketType::Categorical(64u16));
+        assert_ok!(Parimutuel::<T>::on_activation(&market_id).result);
+
+        for asset_idx in 0..64 {
+            let asset = ParimutuelShareOf::<T>::Share(Zero::zero(), asset_idx).into();
+            assert!(T::AssetCreator::asset_exists(asset));
+        }
+
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            market.status = MarketStatus::Resolved;
+            let resolved_outcome = OutcomeReport::Categorical(0u16);
+            market.resolved_outcome = Some(resolved_outcome);
+            Ok(())
+        })?;
+
+        #[block]
+        {
+            Parimutuel::<T>::on_resolution(&market_id);
+        }
+
+        #[cfg(test)]
+        {
+            use frame_support::{pallet_prelude::Weight, traits::OnIdle};
+
+            crate::mock::AssetRouter::on_idle(Zero::zero(), Weight::MAX);
+            for asset_idx in 0..64 {
+                let asset = ParimutuelShareOf::<T>::Share(Zero::zero(), asset_idx).into();
+                assert!(!T::AssetCreator::asset_exists(asset));
+            }
+        }
```

Comment:
Why isn't your tests directly enough for this? I would remove this code. I mean it's just for the benchmark tests and not even a test for a specific real runtime path check.

Reply:
It's just another safety guard to ensure that `on_resolution` did indeed incorporate the overhead for deleting the assets. It does not do any harm from my perspective and it's generally encouraged to verify the state after the benchmark.